rama-http/src/layer/header_from_str_config.rs

```diff
@@ -164,9 +170,12 @@ where
                             .map(|x| x.parse::<T>().map_err(Into::into))
                     })
                 })
-                .collect();
-            let values = result?;
-            if values.is_empty() {
+                .peekable();
+
+            let is_empty = parsed_values.peek().is_none();
+            let values = parsed_values.collect::<Result<C, _>>()?;
```

Comment:
Only collect it right before you insert it into the context. Not needed nor desired to do this upfront.